src/data_request_spec.rs

```diff
@@ -1,49 +1,146 @@
-                            .with_context(|| "Failed to find metric")?
-                            .into(),
-                    );
+                MetricSpec::Metric(metric) => {
+                    let metrics = catalogue.expand_wildcard_metric(metric)?;
+                    let metric_options = catalogue
+                        .get_possible_metric_details(&metrics)
+                        .with_context(|| "Failed to find metric")?;
+                    possible_metrics = if let Some(prev) = possible_metrics {
+                        Some(prev.vstack(&metric_options)?)
+                    } else {
+                        Some(metric_options)
+                    };
                 }
                 MetricSpec::DataProduct(_) => todo!("unsupported metric spec"),
             }
         }
-        Ok(metric_requests)
-    }
 
-    pub fn geom_details(&self, catalogue: &SourceDataRelease) -> Result<String> {
-        Ok(catalogue.geography_file.clone())
+        // Extract the possible matches for the given metrics
+        let possible_metrics = possible_metrics.context("Failed to find matching metrics")?;
+
+        // If a geometry level is specified filter out only those metrics with the level
+        let filtered_possible_metrics = if let Some(level) = &self.geometry.geometry_level {
+            possible_metrics
+                .lazy()
+                .filter(col("geometry_level").eq(lit(level.clone())))
+        } else {
+            possible_metrics.lazy()
+        };
+
+        // If a year is specifed, filter out only years that have that level.
+        let filtered_possible_metrics = if let Some(years) = &self.years {
+            let year_series = Series::new("year_filter", years.clone());
+            filtered_possible_metrics.filter(col("year").is_in(lit(year_series)))
+        } else {
+            filtered_possible_metrics
+        };
+
+        let filtered_possible_metrics = filtered_possible_metrics.collect()?;
+
+        // Iterate through the results and generate the metric requests
+        let metric_requests: Vec<MetricRequest> = filtered_possible_metrics
+            .column("parquet_metric_id")?
+            .str()?
+            .into_iter()
+            .zip(
+                filtered_possible_metrics
+                    .column("parquet_metric_file")?
+                    .str()?,
+            )
+            .filter_map(|(column, file)| {
+                if let (Some(column), Some(file)) = (column, file) {
+                    Some(MetricRequest {
+                        column: column.to_owned(),
+                        file: file.to_owned(),
+                    })
+                } else {
+                    None
+                }
+            })
+            .collect();
+
+        let selected_geometry = filtered_possible_metrics
+            .column("geometry_level")?
+            .str()?
+            .get(0)
+            .expect("Should have geometry")
```

Comment:
Convert to anyhow for now?
```suggestion
            .ok_or(anyhow!("Filtered possible metrics does not contain 'geometry_level'"))?
```